[JIT] Refactor SR invocation of fusion

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70656
* #70464
* #70871
* __->__ #70508
* #70463

We can create the code object at compile time instead or runtime to speed it up. This also makes unnecessary the compilation cache. TODO: figure out if theres a way to cache InterpreterState object

Differential Revision: [D33458648](https://our.internmc.facebook.com/intern/diff/D33458648)

Commit:
Refactor SR invocation of fusion

[ghstack-poisoned]

navahgar commented on torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp:
```diff
@@ -388,21 +388,21 @@ RegisterOperators reg_guard({
 Operation createTensorExprDynamicGroup(const Node* node) {
   auto graph = node->g(attr::Subgraph);
+  Code code(graph, "");
   // This implementation creates a Code object and InterpreterState on every
```
Update this comment to reflect this change.

navahgar reviewed (approved):
Thanks for these changes.

Commit:
Update on "[JIT] Refactor SR invocation of fusion"


We can create the code object at compile time instead or runtime to speed it up. This also makes unnecessary the compilation cache. TODO: figure out if theres a way to cache InterpreterState object

[ghstack-poisoned]

Commit:
Update on "[JIT] Refactor SR invocation of fusion"


We can create the code object at compile time instead or runtime to speed it up. This also makes unnecessary the compilation cache. TODO: figure out if theres a way to cache InterpreterState object

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458648).

Commit:
Update on "[JIT] Refactor SR invocation of fusion"


We can create the code object at compile time instead or runtime to speed it up. This also makes unnecessary the compilation cache. TODO: figure out if theres a way to cache InterpreterState object

Differential Revision: [D33458648](https://our.internmc.facebook.com/intern/diff/D33458648)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458648).

Commit:
Update on "[JIT] Refactor SR invocation of fusion"


We can create the code object at compile time instead or runtime to speed it up. This also makes unnecessary the compilation cache. TODO: figure out if theres a way to cache InterpreterState object

Differential Revision: [D33458648](https://our.internmc.facebook.com/intern/diff/D33458648)

[ghstack-poisoned]

Commit:
Update on "[JIT] Refactor SR invocation of fusion"


We can create the code object at compile time instead or runtime to speed it up. This also makes unnecessary the compilation cache. TODO: figure out if theres a way to cache InterpreterState object

Differential Revision: [D33458648](https://our.internmc.facebook.com/intern/diff/D33458648)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458648).
